.github/workflows/ci.yml

```diff
@@ -40,12 +35,9 @@ jobs:
             ${{ runner.os }}-build-${{ env.cache-name }}-
             ${{ runner.os }}-build-
             ${{ runner.os }}-
-
       - name: Install NPM dependencies
         run: npm install
-
       - name: Build the project
         run: npm run build
-
```

Comment:
I know this is a bit nit-picky, but would you mind preserving the newlines between commands? I personally find it much easier to read that way (I can more easily chunk which commands go together).

Reply:
Hm... I'm not sure. I'm using [yq](https://github.com/mikefarah/yq) to update the yaml files from the command line. So, that's probably being done by that program.

Reply:
This is a bit annoying: https://github.com/mikefarah/yq/issues/465

I'd really like to find a way to just replace

```
        uses: purescript-contrib/setup-purescript@main
```

with

```
        uses: purescript-contrib/setup-purescript@main
        with:
          purescript: 0.14.0-rc3
```

without otherwise changing the files.

Reply:
Sed only works on one line at a time. Otherwise, I would use that. I'm not familiar with `awk`, so I'm not sure whether hta twould be the right tool ot use for this.

We could also use a combination of `head` and `tail`. For example, this gets me the line number that the 'uses:' part starts on:
```bash
cat .github/workflows/ci.yml | grep 'uses: purescript-contrib/setup-purescript' -n | cut -d ':' -f1
```

Reply:
If we know the line that contains `uses: purescript-contrib/setup-purescript`, then we could always insert with sed. Assuming it's on line 10:

```sh
sed -i '' '10i\        with' .github/workflows/ci.yml
sed -i '' '11i\          purescript = "0.14.3-rc" .github/workflows/ci.yml
```

to insert `with` at line 10 and then `purescript ...` at line 11.

Reply:
Nevermind. Sed will work. I forgot that you can replace a single line with multiple lines.

Reply:
I went with this command. It'll work as long as the indentation is the same throughout all files. I'm sure there's a way to account for that properly, but I don't think it's worth the time to fix that until the problem happens:
```
sed -i 's/        uses: purescript-contrib\/setup-purescript@main/        uses: purescript-contrib\/setup-purescript@main\n        with:\n          purescript: "0.14.0-rc3"/' .github/workflows/ci.yml
```